client/lib/i18n-utils/switch-locale.js

```diff
@@ -18,13 +18,14 @@ const getPromises = {};
 /**
  * De-duplicates repeated GET fetches of the same URL while one is taking place.
  * Once it's finished, it'll allow for the same request to be done again.
+ *
  * @param {string} url The URL to fetch
  *
  * @returns {Promise} The fetch promise.
  */
 function dedupedGet( url ) {
 	if ( ! ( url in getPromises ) ) {
-		getPromises[ url ] = fetch( url ).finally( () => delete getPromises[ url ] );
+		getPromises[ url ] = window.fetch( url ).finally( () => delete getPromises[ url ] );
```

Comment:
Does this method ever run on the server? We have a polyfill for `fetch` in node, so adding `window.` here could make any existing server-side usage stop working.

Reply:
This particular `fetch` never runs on the server, but you have a good point. In #37953, I made referencing globals directly illegal -- it's now an ESLint offense. That's because there are too many (hundreds) globals defined on `window` and `document` and ESLint was failing to report real bugs.

That's why we need `window.fetch` or `global.fetch` now. Is there a "global" identifier that would work both in browser and in Node.js? Is `global` that universal global? Is CoreJS or webpack or Babel polyfilling that? I don't know and it would be nice to figure it out.

Reply:
`global` doesn't work on the browser and `window` doesn't work on node. And even on the browser, `window` doesn't work anywhere where there isn't a window, e.g. service workers.

The solution to this is [`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) (stage 4), which should work everywhere. I'm not sure if our pipeline supports it already, and I'm finding it difficult to figure out the state of it in Node as well.

Reply:
Looks like `globalThis` needs `core-js` 3 in Babel, unfortunately 😞

Reply:
There are polyfills we could use in the meantime, however.

Reply:
FYI, the lint change seems to be adding a bunch of noise. Basic functions like `btoa` and constructors like `new Image()` now requiring adding a top-level scope specifier.

I'm doing the following for now, but I really don't like the solution, as it's verbose and doesn't cover e.g. workers:

```js
// TODO: Replace with `globalThis` once we can use it.
const topScope = typeof window !== 'undefined' ? window : global;

new topScope.Image().src = imgUrl;
```

Reply:
I'll add the above comment to the original PR, for reference.